pywr/parameters/_parameters.pyx

```diff
@@ -1078,7 +1103,7 @@ cdef class RecorderThresholdParameter(IndexParameter):
 
     cpdef double value(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
         """Returns a value from the values attribute, using the index"""
-        cdef int ind = self.index(timestep, scenario_index)
+        cdef int ind = self.index(timestep, scenario_index) # don't use get_index!
```

Comment:
If we swapped the order that `get_values`/`get_indices` get called in `before` then we could.